### PR TITLE
chore: Simplify Release Procedures RFD

### PR DESCRIPTION
**Pull request by alilleybrinker**
This RFD recommends changes to simplify the release processes for Hipcheck, explaining the problems with the current setup, and the benefits of the proposed changes.

### CODE REVIEW

File: site/content/rfds/0007-simplified-release-procedures.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+  - Crates.io
+- Docker image:
+  - Docker Hub
+
+When we cut a new release of `hc`, the flow has been:
+
+1. Update the `CHANGELOG.md` for the project with release notes for the new
+   version, open a PR, and get that merged.
+2. Run `cargo release` locally to make a new release of `hc` to Crates.io.
+   At the end of that process, `cargo release` creates a publishes a tag for
```
Comment (j-lanson) [inline site/content/rfds/0007-simplified-release-procedures.md:32]: `creates a publishes a` --> `creates and publishes a`

File: site/content/rfds/0007-simplified-release-procedures.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+2. Tag the new version as `hipcheck-v<NEW_VERSION>`, with `<NEW_VERSION>`
+   replaced with the SemVer version we're releasing (`X.Y.Z`).
+
+That's it! Then, `cargo-dist` would build our release artifacts,
+including the pre-built binaries and the install scripts, and make the
+GitHub Release when it's done.
+
+Once the GitHub Release is made, the Docker build will run, building
+a Docker container to publish with the version of the container image's
+`curl | sh` install script set to the just-release version. Since this
```
Comment (j-lanson) [inline site/content/rfds/0007-simplified-release-procedures.md:81]: `just-release` --> `just-released`

File: site/content/rfds/0007-simplified-release-procedures.md (hunk 1 above)
Comment (alilleybrinker) [reply site/content/rfds/0007-simplified-release-procedures.md:32]: Fixed!

**Review by j-lanson — APPROVED** [review]
I think this makes a lot of sense, and the trade-off seems very worthwhile.

**Comment by alilleybrinker** [comment]
@j-lanson can I get a re-review?

**Review by j-lanson — APPROVED** [review]

**Pull request merged**